### ISSUE TITLE
make check failure to build some test programs under Cygwin and Mingw

### ISSUE THREAD

**Issue by BrianInglis** [libssh2/libssh2#1080]
**Describe the bug**
make check failure to build some test programs under Cygwin and Mingw

**To Reproduce**
`make check`
[cygwin-x86_64-libssh2-1.11.0-1-check.log](https://github.com/libssh2/libssh2/files/11644340/cygwin-x86_64-libssh2-1.11.0-1-check.log)
[mingw64-x86_64-libssh2-1.11.0-1-check.log](https://github.com/libssh2/libssh2/files/11644341/mingw64-x86_64-libssh2-1.11.0-1-check.log)

**Expected behavior**
Test programs build and run, not necessarily successfully. 

**Version (please complete the following information):**
 - Cygwin 3.4.6 under Windows 10 [20]21H2 19044
 - libssh2 version: 1.11.0
 - crypto backend and version: OpenSSL 1.1.1t  7 Feb 2023

**Additional context**
Cygwin build:
```
configure: summary of build options:

  version:          1.11.0
  Host type:        x86_64-pc-cygwin
  Install prefix:   /usr
  Compiler:         gcc
  Compiler flags:   -ggdb -O2 -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=4 -fdebug-prefix-map=$HOME/src/cygwin/libssh2/libssh2-1.11.0-1.x86_64/build=/usr/src/debug/libssh2-1.11.0-1 -fdebug-prefix-map=$HOME/src/cygwin/libssh2/libssh2-1.11.0-1.x86_64/src/libssh2-1.11.0=/usr/src/debug/libssh2-1.11.0-1
  Library types:    Shared=yes, Static=no
  Crypto library:   OpenSSL
  zlib compression: yes
  Clear memory:     yes
  Debug build:      no
  Build examples:   yes
  Run Docker tests: yes
  Run sshd tests:   yes
  Path to sshd:     /usr/sbin/sshd (only for self-tests)
```
Mingw build:
```
configure: summary of build options:
  version:          1.11.0
  Host type:        x86_64-w64-mingw32
  Install prefix:   /usr/x86_64-w64-mingw32/sys-root/mingw
  Compiler:         x86_64-w64-mingw32-gcc
  Compiler flags:   -ggdb -O2 -pipe -Wall -Werror=format-security -fdebug-prefix-map=$HOME/src/cygwin/libssh2/mingw64-x86_64-libssh2/mingw64-x86_64-libssh2-1.11.0-1.noarch/build=/usr/src/debug/mingw64-x86_64-libssh2-1.11.0-1 -fdebug-prefix-map=$HOME/src/src/cygwin/libssh2/mingw64-x86_64-libssh2/mingw64-x86_64-libssh2-1.11.0-1.noarch/src/libssh2-1.11.0=/usr/src/debug/mingw64-x86_64-libssh2-1.11.0-1
  Library types:    Shared=yes, Static=no
  Crypto library:   Windows Cryptography API: Next Generation
  zlib compression: yes
  Clear memory:     yes
  Debug build:      no
  Build examples:   yes
  Run Docker tests: yes
  Run sshd tests:   yes
  Path to sshd:     /usr/sbin/sshd (only for self-tests)
```

**Comment by willco007** [comment]
Dupe of #1056?

**Comment by vszakats** [comment]
Duplicate of #1056

Plus this warning with Cygwin that we haven't seen yet:
```
In file included from $HOME/src/cygwin/libssh2/libssh2-1.11.0-1.x86_64/src/libssh2-1.11.0/tests/openssh_fixture.c:57:
$HOME/src/cygwin/libssh2/libssh2-1.11.0-1.x86_64/src/libssh2-1.11.0/tests/openssh_fixture.c: In function 'run_command_varg':
$HOME/src/cygwin/libssh2/libssh2-1.11.0-1.x86_64/src/libssh2-1.11.0/tests/openssh_fixture.c:136:37: warning: array subscript has type 'char' [-Wchar-subscripts]
  136 |         while(end > 0 && isspace(buf[end - 1])) {
      |                                  ~~~^~~~~~~~~
```

**Comment by BrianInglis** [comment]
Duplicate of #1056

**Comment by vszakats** [comment]
Cygwin CI tests landed in f1e96e733fefb495bc31b07f5c2a5845ff877c9c. It also catches the warning above: https://github.com/libssh2/libssh2/actions/runs/5869856823/job/15915700574#step:6:108
